modular_splurt/code/datums/components/pregnancy.dm

```diff
@@ -0,0 +1,482 @@
+
+	eggo.TakeComponent(src)
+	eggo.forceMove(location)
+
+	return TRUE
+
+//not how genetics work but okay
+/datum/component/pregnancy/proc/determine_baby_dna(mob/living/carbon/human/babby)
+	if(mother_dna && father_dna)
+		mother_dna.transfer_identity_random(father_dna, babby)
+	else if(mother_dna && !father_dna)
+		mother_dna.transfer_identity_random(babby.dna, babby)
+	else if(!mother_dna && father_dna)
+		father_dna.transfer_identity_random(babby.dna, babby)
+
+/datum/component/pregnancy/proc/determine_baby_features(mob/living/carbon/human/babby)
+
+	var/list/final_features = list()
+
+	transfer_randomized_list(final_features, mother_features, father_features)
+
+	if(final_features["skin_tone"])
+		babby.skin_tone = final_features["skin_tone"]
+	if(final_features["hair_color"])
+		babby.hair_color = final_features["hair_color"]
+	if(final_features["facial_hair_color"])
+		babby.facial_hair_color = final_features["facial_hair_color"]
+	if(final_features["left_eye_color"])
+		babby.left_eye_color = final_features["left_eye_color"]
+	if(final_features["right_eye_color"])
+		babby.right_eye_color = final_features["right_eye_color"]
+
+	babby.hair_style = pick("Bedhead", "Bedhead 2", "Bedhead 3")
+	babby.facial_hair_style = "Shaved"
+	babby.underwear = "Nude"
+	babby.undershirt = "Nude"
+	babby.socks = "Nude"
+
+	babby.saved_underwear = babby.underwear
+	babby.saved_undershirt = babby.undershirt
+	babby.saved_socks = babby.socks
+
+/datum/component/pregnancy/proc/generic_pragency_start()
+	if(revealed)
+		carrier.apply_status_effect(/datum/status_effect/pregnancy)
+	if(ishuman(carrier))
+		human_pragency_start(carrier)
+	ADD_TRAIT(carrier, TRAIT_PREGNANT, PREGNANCY_TRAIT)
+
+/datum/component/pregnancy/proc/generic_pragency_end()
+	REMOVE_TRAIT(carrier, TRAIT_PREGNANT, PREGNANCY_TRAIT)
+	carrier.remove_status_effect(/datum/status_effect/pregnancy)
+	if(ishuman(carrier))
+		human_pragency_end(carrier)
+
+/datum/component/pregnancy/proc/human_pragency_start(mob/living/carbon/human/gregnant)
+	if(pregnancy_inflation)
+		//give them a king ass ripper belly initially
+		var/obj/item/organ/genital/belly/belly = gregnant.getorganslot(ORGAN_SLOT_BELLY)
+		if(!belly)
+			belly = gregnant.give_genital(/obj/item/organ/genital/belly)
+		if(added_size > 0)
+			belly.modify_size(added_size)
+	if(pregnancy_breast_growth)
+		var/obj/item/organ/genital/breasts/boob = gregnant.getorganslot(ORGAN_SLOT_BREASTS)
+		if(!boob)
+			boob = gregnant.give_genital(/obj/item/organ/genital/breasts)
+		if(added_size > 0)
+			boob.modify_size(added_size)
+	return TRUE
+
+/datum/component/pregnancy/proc/human_pragency_end(mob/living/carbon/human/gregnant)
+	//get rid of king ass ripper belly
+	var/obj/item/organ/genital/belly/belly = gregnant.getorganslot(ORGAN_SLOT_BELLY)
+	var/obj/item/organ/genital/breasts/boob = gregnant.getorganslot(ORGAN_SLOT_BREASTS)
+	if(pregnancy_inflation)
+		belly?.modify_size(-added_size)
+	if(pregnancy_breast_growth)
+		boob?.modify_size(-added_size)
```

Comment:
```suggestion
	if(pregnancy_breast_growth)
		added_size = 0
```
I don't think breasts should reduce their size here giving they'd realistically be still full of milk to feed their spawn, maybe do set the added size to 0 however so it doesn't cause boob growth when shoved back into someone for incubation till it goes up a stage inside them